### ISSUE TITLE
Update to Swift 2

### ISSUE THREAD

**Issue by ryanmasondavies** [iotize/FaceDetection#7]
Have gone through the basic migration process to support Swift 2, but haven't fully embraced the improvements. Code should be refactored to use newer syntax such as `guard`, `defer`, where needed, and use `struct` and protocols in place of classes where possible.

**Comment by digitallysavvy** [comment]
I just tried your current build on an iPhone 6 using iOS 9.3 and the entire camera image is pixilated, not just the face. I created a new pull request with updated code to work with Swift 2.3. It uses the black circle over the face.
![img_2921](https://cloud.githubusercontent.com/assets/6816583/14069525/37042172-f469-11e5-88b8-e7485a81e69a.jpg)
![img_2916](https://cloud.githubusercontent.com/assets/6816583/14069528/4bfb58de-f469-11e5-899c-2b12b29338ef.jpg)
